Gems/Atom/RPI/Code/Source/RPI.Reflect/Image/StreamingImageAsset.cpp

```diff
@@ -124,5 +213,45 @@ namespace AZ
 
             return mipChainAsset->GetSubImageData(mip - mipChain.m_mipOffset, slice);
         }
+
+        template<>
+        float StreamingImageAsset::GetSubImagePixelValue<float>(uint32_t mip, uint32_t slice, uint32_t x, uint32_t y, uint32_t componentIndex)
+        {
+            // TODO: Use the component index
+            (void)componentIndex;
+
+            auto imageData = GetSubImageData(mip, slice);
+
+            if (!imageData.empty())
+            {
+                const AZ::RHI::ImageDescriptor imageDescriptor = GetImageDescriptor();
+                auto width = imageDescriptor.m_size.m_width;
+                size_t index = (y * width) + x;
+
+                return Internal::RetrieveFloatValue(imageData.data(), index, imageDescriptor.m_format);
+            }
+
+            return 0.0f;
+        }
```

Comment:
The thing to think about with this API is that if we're looking up 1 million pixel values, a lot of this will be redundant work.  Is there a way to design the API (or create a second one) to avoid the redundancies for doing lots of lookups in a row?

Reply:
I think we should have a region based API in addition to single values to take advantage of those speedups when possible. I'm find with leaving that till later though to get something working as fast as possible.

Reply:
What if we changed it to be like:

```
void GetSubImagePixelValue(AZStd::array_view<AZ::Vector2> positions, AZStd::array_view<float> outValues, uint32_t componentIndex = 0, uint32_t mip = 0, uint32_t slice = 0)
```

We could leave the single pixel gettr too, but like you said it seems like the more practical case would be when you're interested in querying multiple values.

Reply:
I've added a region based API as we discussed offline, will update the PR shortly